## **Chapter 9 - Exception Handling**
This folder contains the artifacts for the chapter 9 of the the "Python Fundamentals" book. This chapter involves 
the creation of a code challenge about **exception handling**. 

A more detailed document for this project can be found [here](#).

## **Project Description**
                          
                                  
The solution code for this challenge can be found [here](https://github.com/luizhenriqueds/packt-courseware/blob/master/projects/chapter-9/validate_email_utils.py).

## **Take home extension**

The solution code for the extended challenge can be found at the following addresses:
[part 1](https://github.com/luizhenriqueds/packt-courseware/blob/master/projects/chapter-9/validate_email_utils_extended.py)
and [part 2](https://github.com/luizhenriqueds/packt-courseware/blob/master/projects/chapter-9/validate_email_utils_extended_finally.py).

## **Optional exercise**

The solution code for the optional challenge can be found [here](https://github.com/luizhenriqueds/packt-courseware/blob/master/projects/chapter-9/validate_email_utils_optional_regex.py).

## **Unit Testing**

To run the tests, just execute the test.py file in our preferred IDE. To run tests separately, select the test you
wish to run, click the right button, then "Run Unittests in test.py" in pycharm. 

To run the test from command line, cd into the project's folder, then run the command: ``python test.py``.

The unit test for the code challenges can be found [here](#).
